cmd/plume/fedora.go

```diff
@@ -0,0 +1,120 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+)
+
+var (
+	specArch            string
+	specComposeID       string
+	specEnv             string
+	specRespin          string
+	specImageType       string
+	specTimestamp       string
+	specFedoraVersion   string
+	specFedoraBoard     string
+	awsFedoraBoards     = []string{"amd64-usr"}
+	awsFedoraPartitions = []awsPartitionSpec{
+		awsPartitionSpec{
+			Name:         "AWS",
+			Profile:      "default",
+			Bucket:       "fedora-s3-bucket-fedimg-test",
+			BucketRegion: "us-east-1",
+			LaunchPermissions: []string{
+				"013116697141",
+			},
+			Regions: []string{
+				"us-east-1",
+				"us-east-2",
+				"us-west-1",
+				"us-west-2",
+			},
+		},
+	}
+
+	fedoraSpecs = map[string]channelSpec{
+		"rawhide": channelSpec{
+			BaseURL: "https://koji.fedoraproject.org/compose/rawhide",
+			System:  "Fedora",
+			AWS: awsSpec{
+				BaseName:        "Fedora",
+				BaseDescription: "Fedora AMI",
+				Prefix:          "fedora_{{.Env}}_ami_",
+				Image:           "Fedora-AtomicHost-{{.Version}}-{{.Timestamp}}.{{.Respin}}.{{.Arch}}.raw.xz",
+				Partitions:      awsFedoraPartitions,
+			},
+		},
+		"updates": channelSpec{
+			BaseURL: "https://koji.fedoraproject.org/compose/updates",
+			System:  "Fedora",
+			AWS: awsSpec{
+				BaseName:        "Fedora",
+				BaseDescription: "Fedora AMI",
+				Prefix:          "fedora_{{.Env}}_ami_",
+				Image:           "Fedora-AtomicHost-{{.Version}}-{{.Timestamp}}.{{.Respin}}.{{.Arch}}.raw.xz",
+				Partitions:      awsFedoraPartitions,
+			},
+		},
+		"branched": channelSpec{
+			BaseURL: "https://koji.fedoraproject.org/compose/branched",
+			System:  "Fedora",
+			AWS: awsSpec{
+				BaseName:        "Fedora",
+				BaseDescription: "Fedora AMI",
+				Prefix:          "fedora_{{.Env}}_ami_",
+				Image:           "Fedora-{{.ImageType}}-{{.Version}}-{{.Timestamp}}.n.{{.Respin}}.{{.Arch}}.raw.xz",
+				Partitions:      awsFedoraPartitions,
+			},
+		},
+		"cloud": channelSpec{
+			BaseURL: "https://koji.fedoraproject.org/compose/cloud",
+			System:  "Fedora",
+			AWS: awsSpec{
+				BaseName:        "Fedora",
+				BaseDescription: "Fedora AMI",
+				Prefix:          "fedora_{{.Env}}_ami_",
+				Image:           "Fedora-{{.ImageType}}-{{.Version}}-{{.Timestamp}}.{{.Respin}}.{{.Arch}}.raw.xz",
+				Partitions:      awsFedoraPartitions,
+			},
+		},
+	}
+)
+
+func AddFedoraSpecFlags(flags *pflag.FlagSet) {
+	flags.StringVarP(&specEnv, "environment", "E", "prod", "instance environment")
+	flags.StringVarP(&specImageType, "imagetype", "I", "Cloud-Base", "type of image")
+	flags.StringVarP(&specFedoraVersion, "fedoraversion", "F", "29", "fedora release version")
```

Comment:
I too +1 on this, but I could not come around with a solution to avoid this, as there were too many permutations & combinations. Another way would have been to pass all the names & URLs from the script that would call plume, but that would lead to a bunch of work on script side, and also removing the relevant code from this PR.

Reply:
I'm not sure I understand, I meant this exclusively towards things that had actual duplicates in the existing set (like `version` & `fedoraversion` or `board` & `arch`).

Are you trying to make it so your script just calls plume without arguments?

Reply:
@arithx @bgilbert @dustymabe So do we go here with the current set of arguments now, and later change them?